packages/middleware-sdk-s3/src/use-regional-endpoint.ts

```diff
@@ -0,0 +1,43 @@
+import { HttpRequest } from "@aws-sdk/protocol-http";
+import {
+  BuildHandler,
+  BuildHandlerArguments,
+  BuildHandlerOptions,
+  BuildHandlerOutput,
+  BuildMiddleware,
+  MetadataBearer,
+  Pluggable,
+  Provider,
+} from "@aws-sdk/types";
+
+type PreviouslyResolved = {
+  region: Provider<string>;
+  isCustomEndpoint: boolean;
+};
+
+export function useRegionalEndpointMiddleware(config: PreviouslyResolved): BuildMiddleware<any, any> {
+  return <Output extends MetadataBearer>(next: BuildHandler<any, Output>): BuildHandler<any, Output> => async (
+    args: BuildHandlerArguments<any>
+  ): Promise<BuildHandlerOutput<Output>> => {
+    const { request } = args;
+    if (!HttpRequest.isInstance(request) || config.isCustomEndpoint) return next({ ...args });
+    if (request.hostname === "s3.amazonaws.com") {
+      request.hostname = "s3.us-east-1.amazonaws.com";
+    } else if ("aws-global" === (await config.region())) {
+      request.hostname = "s3.amazonaws.com";
+    }
+    return next({ ...args });
+  };
+}
```

Comment:
Can be changed to an arrow function
```suggestion
export const useRegionalEndpointMiddleware = (config: PreviouslyResolved): BuildMiddleware<any, any> => <
  Output extends MetadataBearer
>(
  next: BuildHandler<any, Output>
): BuildHandler<any, Output> => async (args: BuildHandlerArguments<any>): Promise<BuildHandlerOutput<Output>> => {
  const { request } = args;
  if (!HttpRequest.isInstance(request) || config.isCustomEndpoint) return next({ ...args });
  if (request.hostname === "s3.amazonaws.com") {
    request.hostname = "s3.us-east-1.amazonaws.com";
  } else if ("aws-global" === (await config.region())) {
    request.hostname = "s3.amazonaws.com";
  }
  return next({ ...args });
};
```

Reply:
Oops, I copy & pasted from other middleware and didn't pay attention.